cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/cache/chart_cache.go

```diff
@@ -329,6 +329,47 @@ func (c *ChartCache) DeleteChartsForRepo(repo *types.NamespacedName) error {
 	return nil
 }
 
+func (c *ChartCache) DeleteChartsForRepo(repo *types.NamespacedName) error {
+	log.Infof("+DeleteChartsForRepo(%s)", repo)
+	defer log.Infof("-DeleteChartsForRepo(%s)", repo)
+
+	return c.deleteChartsHelper(repo, sets.String{})
+}
+
+func (c *ChartCache) PurgeObsoleteChartVersions(keepThese []models.Chart) error {
```

Comment:
Just reading the code below, I think `keepThese` is the slice of charts for which you're wanting to keep all versions? (as in, below, each chart is iterated with the redis key for each version added to the set of keys to keep for that repo, which is then passed as `keep` to `deleteChartsHelper`).

I must be missing something here, because it looks like it only ever calls `deleteChartsHelper` with repos of charts that were in the `keepThese` slice? I expected to find a function that purges all *other* repositories, other than the ones you wanted to keep (maybe it does and I'm just mis-reading).

EDIT: Oh - reading further down, I think I understand: this function is called when re-importing charts after an update to the repo, for example, so `keepThese` is actually populated from the new data, meaning that if the new data no longer includes a certain version, it'll get purged here. OK, I think I was confused initially because it wasn't clear to me that `keepThese` was the new data that hadn't been synced yet. (?)

Reply:
yes, you got it